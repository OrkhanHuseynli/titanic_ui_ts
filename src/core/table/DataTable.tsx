import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import {Typography} from "@material-ui/core";

const useStyles = makeStyles({
    root: {
        width: '100%',
        overflowX: 'auto',
    },
    table: {
        minWidth: 350,
    },
});

export type DataTableProps = {
    dataRows: DataTableRow[]
}

type DataTableRow = {
    name: string
    value: string
}
const DataTable:React.FC<DataTableProps> = (props: DataTableProps) => {
    const classes = useStyles();

    return (
        <Paper className={classes.root}>
            <Table className={classes.table} aria-label="data table">
                <TableHead>
                    <TableRow>
                        <TableCell>Data info</TableCell>
                        <TableCell align="right">Value</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>
                    {props.dataRows.map(row => (
                        <TableRow key={row.name}>
                            <TableCell component="th" scope="row">
                                {row.name}
                            </TableCell>
                            <TableCell align="right"><Typography variant="body1" gutterBottom>{row.value}</Typography></TableCell>
                        </TableRow>
                    ))}
                </TableBody>
            </Table>
        </Paper>
    );
}

export default DataTable;