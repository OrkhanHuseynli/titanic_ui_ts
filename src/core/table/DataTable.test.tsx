import {configure, shallow, ShallowWrapper} from "enzyme";
import { create, ReactTestRenderer } from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16'

import * as React from "react";
import DataTable, {DataTableProps} from "./DataTable";

let wrapper: ShallowWrapper<DataTableProps>;
let snapshot: ReactTestRenderer;

beforeEach(() => {
    const mockDataTableRows = [{name: "data", value: "value"}];
    const mockElement =<DataTable dataRows={mockDataTableRows}/>;
    wrapper = shallow(mockElement);
    snapshot = create(mockElement);
});

configure({adapter: new Adapter()});

describe('DataTable', () => {
    test('it matches the snapshot', () => {
        expect(snapshot.toJSON()).toMatchSnapshot();
    });
});