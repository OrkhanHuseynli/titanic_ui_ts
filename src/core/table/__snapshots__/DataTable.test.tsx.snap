// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable it matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
>
  <table
    aria-label="data table"
    className="MuiTable-root makeStyles-table-2"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Data info
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          scope="row"
        >
          data
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            value
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
